### PR TITLE
[coordinator] Add metric for writes that have empty annotation

### PR DESCRIPTION
**Pull request by vpranckaitis**
**What this PR does / why we need it**:

Adds coordinator metric for the count of written datapoints that have empty annotation. 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/cmd/services/m3coordinator/ingest/write.go [hunk 1]
```diff
@@ -358,6 +375,20 @@ func (d *downsamplerAndWriter) writeToStorage(
...
+func (d *downsamplerAndWriter) maybeRecordDatapointsWithEmptyAnnotation(q *storage.WriteQuery) {
+	if len(q.Annotation()) == 0 {
+		n := int64(len(q.Datapoints()))
+		switch q.Attributes().MetricsType {
+		case storagemetadata.UnaggregatedMetricsType:
+			d.metrics.unaggregatedDatapointsWithEmptyAnnotation.Inc(n)
+		case storagemetadata.AggregatedMetricsType:
+			d.metrics.aggregatedDatapointsWithEmptyAnnotation.Inc(n)
+		default:
+			// ignore
```
Comment (vpranckaitis) [inline src/cmd/services/m3coordinator/ingest/write.go:387]: There are other types, though I'm not sure if those should be taken into account

https://github.com/m3db/m3/blob/d95a5dfa41ec02af603bfce3e421ac473d11dce8/src/query/storage/m3/storagemetadata/types.go#L32-L40

File: src/cmd/services/m3coordinator/ingest/write.go [hunk 2]
```diff
@@ -124,13 +126,26 @@ func NewDownsamplerAndWriter(
 	workerPool xsync.PooledWorkerPool,
 	instrumentOpts instrument.Options,
 ) DownsamplerAndWriter {
-	scope := instrumentOpts.MetricsScope().SubScope("downsampler")
+	var (
+		scope = instrumentOpts.MetricsScope().SubScope("downsampler")
+
+		emptyAnnotationMetricName = "datapoints_with_empty_annotation"
```
Comment (linasm) [inline src/cmd/services/m3coordinator/ingest/write.go:132]: Nit: maybe let's shorten this to 
```suggestion
		emptyAnnotationMetricName = "writes_without_annotation"
```
(and all the related symbols, accordingly)

File: src/cmd/services/m3coordinator/ingest/write.go [hunk 3]
```diff
@@ -124,13 +126,26 @@ func NewDownsamplerAndWriter(
...
+		typeTag                   = func(t string) map[string]string {
+			return map[string]string{"type": t}
```
Comment (linasm) [inline src/cmd/services/m3coordinator/ingest/write.go:134]: I'd really like to avoid further overloading of "type" term, this would only add to the confusion as the end goal of these new metrics is to track down the time series that are missing the actual "metrics type". Could we rename this label to "storage-policy" for example (as this value is actually derived from `StoragePolicy` value)?

File: src/cmd/services/m3coordinator/ingest/write.go (hunk 1 above)
Comment (linasm) [reply src/cmd/services/m3coordinator/ingest/write.go:387]: Looks like `storageAttributesFromPolicy` can only return either `UnaggregatedMetricsType` or `AggregatedMetricsType`. If you want to be fully covered, you could add some kind of "storage-policy: other" label for the `default` branch of the `switch` statement (but I'm not advocating for this).

**Review by linasm — APPROVED** [review]
LGTM but have a few suggestions wrt naming.

File: src/cmd/services/m3coordinator/ingest/write.go (hunk 2 above)
Comment (vpranckaitis) [reply src/cmd/services/m3coordinator/ingest/write.go]: https://github.com/m3db/m3/pull/3553/commits/80e0205445535c5b780dcf63e623efc65cf98e27#diff-bd39ac14b90a7e77e48b860a55e1927d98fd3378a61676da3b11c8b4a7698958R132

https://github.com/m3db/m3/pull/3553/commits/f0e4b8a8b5cd6a90883a27d66efd5bb42fac9490

https://github.com/m3db/m3/pull/3553/commits/b3fe075280c69f5393d160a1e272a909f690c5b1

File: src/cmd/services/m3coordinator/ingest/write.go (hunk 3 above)
Comment (vpranckaitis) [reply src/cmd/services/m3coordinator/ingest/write.go]: https://github.com/m3db/m3/pull/3553/commits/80e0205445535c5b780dcf63e623efc65cf98e27#diff-bd39ac14b90a7e77e48b860a55e1927d98fd3378a61676da3b11c8b4a7698958R133-R137

**Comment by vpranckaitis** [comment]
superseded by #3555
